### DISCUSSION TITLE
Completely hide close/minimise/maximise window controls

### DISCUSSION THREAD

**Discussion by peskybogle** [NativePHP/laravel#344] [Ideas]
Allow us to completely hide the window control buttons.

We currently have options to hide the title bar:
->titleBarHidden()

disable the minimise/maximise/close buttons:
->minimizable(false)
->maximizable(false)
->closable(false))

and hide the controls except on hover:
->titleBarButtonsOnHover()

but we have no way of hiding the controls completely.

In a windowed app, it would be helpful to allow complete hiding of the controls so we can implement our own functionality. It allows for a different user experience, and also allows us to do more interesting custom UIs

**Comment by Tray2** [comment]
@funnyoldgames Have you tried using `->frameless()` when opening the window? I guess that should give you what you want.

**Reply by simonhamp** [reply]
Great shout! @funnyoldgames `->frameless()` is definitely what you're after

**Reply by peskybogle** [reply]
@simonhamp So I wrote a long reply to this saying it didn't work and listing all my config stuff, but discovered the issue.

If you use both:

->frameless()
->titleBarHidden()

then you still get the buttons. That's what I had, and I was about to say it wasn't working.

However I worked out that If you only use:

->frameless()

and get rid of:

>titleBarHidden()

then you get what I wanted.

I'm not sure why ->titleBarHidden() brings back the buttons, and now I know it does I can leave it out, but it's not (at least to me as a noob) the most obvious behaviour. But I have it working, so for me it's at least solved. Might be worth documenting though.

Thanks, both!

**Reply by simonhamp** [reply]
Interesting... does the order of the function calls make a difference

**Reply by peskybogle** [reply]
@simonhamp Nope. I wondered the same, but have tried them both ways and got the same result. Perhaps one of my other ones is having a knock-on effect. My full list is:

        Window::open('main')
        ->width(1440)
        ->height(900)
        ->minWidth(1400)
        ->minHeight(900)
        ->resizable(true)
        ->rememberState()
        ->hideMenu()
        ->backgroundColor('#0f1017')
        ->minimizable(false)
        ->maximizable(false)
        ->titleBarHidden() 
        ->frameless()

**Reply by simonhamp** [reply]
Without digging too hard, my guess is that `titleBarHidden` takes precedence

This could be down to Electron or even a macOS thing, probably not something we can/should handle in NativePHP

That said, it's probably worth a note in the docs to indicate the order of precedence in some of these settings and/or some screenshots as a visual guide
